### PR TITLE
make target to promote hotfix build

### PR DESCRIPTION
**Pull request by acookin**
Signed-off-by: Alix Cook <alixcook@datawire.io>

## Description

Make target to make hotfixes easier

## Related Issues
List related issues.

## Testing
Can't really test it, we'll test it when we do the next hotfix :)

## Checklist

 - [ ] I made sure to update `CHANGELOG.md`.

   Remember, the CHANGELOG needs to mention:
    + Any new features
    + Any changes to our included version of Envoy
    + Any non-backward-compatible changes
    + Any deprecations

 - [ ] This is unlikely to impact how Ambassador performs at scale.

   Remember, things that might have an impact at scale include:
    + Any significant changes in memory use that might require adjusting the memory limits
    + Any significant changes in CPU use that might require adjusting the CPU limits
    + Anything that might change how many replicas users should use
    + Changes that impact data-plane latency/scalability

 - [ ] My change is adequately tested.

   Remember when considering testing:
    + LEGACY MODE TESTS DO NOT RUN FOR EVERY PR. If your change is affected by legacy mode, you need
      to run legacy-mode tests manually (set `AMBASSADOR_LEGACY_MODE=true` and run the tests).
      (This will be fixed soon.)
    + Your change needs to be specifically covered by tests.
       + Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
    + You also need to make sure that the _entire area being changed_ has adequate test coverage.
       + If existing tests don't actually cover the entire area being changed, add tests.
       + This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
    + We should lean on the bulk of code being covered by unit tests, but...
    + ... an end-to-end test should cover the integration points

 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.

### CODE REVIEW

File: builder/builder.mk [hunk 1]
```diff
@@ -845,6 +845,59 @@ release/promote-oss/dev-to-passed-ci:
...
+		commit=$$(git rev-parse HEAD) ;\
+		dev_version=$$(aws s3 cp s3://datawire-static-files/dev-builds/$$commit -) ;\
+		if [ -z "$$dev_version" ]; then \
+			printf "$(RED)==> found no dev version for $$commit in S3...$(END)\n" ;\
+			exit 1 ;\
+		fi ;\
+		printf "$(CYN)==> $(GRN)Promoting $(BLU)$$commit$(GRN) => $(BLU)$$dev_version$(GRN) in S3...$(END)\n" ;\
+		echo "$$dev_version" | aws s3 cp - s3://datawire-static-files/passed-pr/$$commit ;\
+	}
+.PHONY: release/promote-oss/dev-to-passed-ci
```
Comment (kflynn) [inline builder/builder.mk:861]: This should be `pr-to-passed-ci`, right? 

Also, the Circle config looks to me like this'll only run on PRs to `master` or to `rel/vX.Y.Z`, right?

**Review by kflynn — CHANGES_REQUESTED** [review]
Two kinda nitpicky questions. 

Overall, though, looks good, thanks!! The idea is that you run `make release/promote-oss/to-hotfix` manually, right?

File: builder/builder.mk (hunk 1 above)
Comment (acookin) [reply builder/builder.mk]: whoops and whoops. fixed both things

**Review by kflynn — APPROVED** [review]
Hit it! and thanks. 🙂

**Comment by kflynn** [comment]
(@acookin confirmed that, yes, the idea is that you run make release/promote-oss/to-hotfix manually.)

**Pull request merged**